src/sample-executors/BaseExecutor.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {Owned} from "solmate/src/auth/Owned.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+
+import {ResolvedOrder, SignedOrder} from "../base/ReactorStructs.sol";
+import {IReactorCallback} from "../interfaces/IReactorCallback.sol";
+import {IReactor} from "../interfaces/IReactor.sol";
+
+library Commands {
+    bytes1 internal constant COMMAND_TYPE_MASK = 0x0f;
+
+    uint256 constant EXECUTE = 0x00;
+    uint256 constant EXECUTE_BATCH = 0x01;
+    uint256 constant PERMIT = 0x02;
+    uint256 constant PERMIT_BATCH = 0x03;
+    // 0x04 - 0x0f are reserved for future use
+}
+
+struct PermitData {
+    address token;
+    bytes data;
+}
+
+abstract contract BaseExecutor is IReactorCallback, Owned {
+    /// @notice Thrown when attempting to execute commands and an incorrect number of inputs are provided
+    error LengthMismatch();
+    error InvalidCommandType(uint256 commandType);
+
+    IReactor public immutable reactor;
+
+    constructor(IReactor _reactor, address _owner) Owned(_owner) {
+        reactor = _reactor;
+    }
+
+    function reactorCallback(ResolvedOrder[] calldata, bytes calldata callbackData) external virtual;
+
+    function _restrictCall() internal virtual {}
```

Comment:
comment on what htis is meant to be used for? assuming its auth, in which case seems like a strange thing to have as a top level virtual function. i.e. maybe some executors want different auth for different commands and so on? this model makes that impossible

Reply:
er why not have it just live as a modifier? this looks like its just a wrapper over onlyWhitelistedCaller

Reply:
Ah yeah will just do super.multicall to apply the whitelistedCaller modifier in SwapRouter02